common/src/main/java/tachyon/exception/TachyonException.java

```diff
@@ -57,19 +54,47 @@ public TachyonTException toTachyonTException() {
     return new TachyonTException(mType.name(), getMessage());
   }
 
+  /**
+   * Constructs a {@link TachyonException} from a {@link TachyonTException}.
+   *
+   * @param e the {link TachyonTException} to convert to a {@link TachyonException}
+   * @return a {@link TachyonException} of the type specified in e, with the message specified in e
+   */
+  public static TachyonException of(TachyonTException e) {
+    TachyonExceptionType exceptionType = TachyonExceptionType.valueOf(e.type);
+    Class<? extends TachyonException> throwClass = exceptionType.getExceptionClass();
+    Exception reflectError;
+    try {
+      TachyonException throwInstance =
+          throwClass.getConstructor(String.class).newInstance(e.getMessage());
+      return throwInstance;
+      // This will be easier when we move to Java 7, when instead of catching Exception we can catch
+      // ReflectiveOperationException, the Java 7 superclass for these Exceptions
+    } catch (InstantiationException e1) {
+      reflectError = e1;
+    } catch (IllegalAccessException e1) {
+      reflectError = e1;
+    } catch (InvocationTargetException e1) {
+      reflectError = e1;
+    } catch (NoSuchMethodException e1) {
+      reflectError = e1;
+    }
+    String errorMessage = "Could not instantiate " + throwClass.getName() + " with a String-only "
+        + "constructor: " + reflectError.getMessage();
+    throw new IllegalStateException(errorMessage);
+  }
+
+  /**
+   * If the given {@link TachyonException} is of the given class, throw it.
```

Comment:
`Checks if the given exception is an instance of the given derived class and if so, downcasts the exception and throws it.`